addons/stock/models/stock_picking.py

```diff
@@ -700,6 +700,8 @@ def action_assign(self):
         also impact the state of the picking as it is computed based on move's states.
         @return: True
         """
+        if self.filtered(lambda picking: picking.state == 'done'):
+            raise UserError(_('You cannot reserve a picking that has been set to "Done".'))
         self.filtered(lambda picking: picking.state == 'draft').action_confirm()
```

Comment:
Do we want to auto-confirm all draft picking in the same time ? (I do see any
It is maybe return a error for draft as for the "done" state ? Or do nothing for both state and don't raise a error ?

Reply:
@ryv-odoo I'm not sure. Why you think confirm all draft picking can be dangerous?

Reply:
After some discussion with arm about that: it is Ok to do that. We can let it like that.

Reply:
Maybe you can merge your commit, it is the same topic.

Reply:
done